### PR TITLE
STM32MP1xx add SPI LL

### PR DESCRIPTION
**Pull request by yboutreux**
CMSIS register modification are cherry-picked from Cube CMSIS v1.1.0.rc1, modifications non-related to SPI hadn't been selected.
Add SPI LL for STM32MP1x SoC.

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Last commit miss update on stm32cube/stm32mp1xx/CMakeLists.txt

**Comment by yboutreux** [comment]
> Last commit miss update on stm32cube/stm32mp1xx/CMakeLists.txt

Updated to add changes on stm32cube/stm32mp1xx/README

File: stm32cube/stm32mp1xx/README [hunk 1]
```diff
@@ -72,3 +72,10 @@ Patch List:
       ext/hal/st/stm32cube/stm32mp1xx/soc/stm32mp157axx_cm4.h
       ext/hal/st/stm32cube/stm32mp1xx/soc/stm32mp157cxx_cm4.h
     ST Bug tracker ID: None
+
+    * Add SPI LL drivers
+    Add SPI LL drivers to cube.
+    Impacted files:
+      ext/hal/st/stm32cube/stm32mp1xx/drivers/include/stm32mp1xx_ll_spi.h
+      ext/hal/st/stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c
+    ST Bug tracker ID: None
```
Comment (erwango) [inline stm32cube/stm32mp1xx/README:93]: @yboutreux , can you give expected release version here?

**Review by erwango — CHANGES_REQUESTED** [review]

File: stm32cube/stm32mp1xx/README (hunk 1 above)
Comment (yboutreux) [reply stm32cube/stm32mp1xx/README:93]: > @yboutreux , can you give expected release version here?

Modification done

**Comment by yboutreux** [comment]
I also added the Cube version inside the commit

File: stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c [hunk 2]
```diff
@@ -0,0 +1,743 @@
...
+  assert_param(IS_I2S_ALL_INSTANCE(SPIx));
+  assert_param(IS_LL_I2S_PRESCALER_LINEAR(PrescalerLinear));
+  assert_param(IS_LL_I2S_PRESCALER_PARITY(PrescalerParity));
+
+  /* Write to SPIx I2SPR */
+  MODIFY_REG(SPIx->I2SCFGR, SPI_I2SCFGR_I2SDIV | SPI_I2SCFGR_ODD, (PrescalerLinear << SPI_I2SCFGR_I2SDIV_Pos) |
+             (PrescalerParity << SPI_I2SCFGR_ODD_Pos));
+}
+
+/**
```
Comment (atorgue) [inline stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c:713]: why those empty lines ?

**Review by atorgue — APPROVED** [review]

File: stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c (hunk 2 above)
Comment (yboutreux) [reply stm32cube/stm32mp1xx/drivers/src/stm32mp1xx_ll_spi.c:713]: Picked directly code generated for CubeMP1

**Comment by yboutreux** [comment]
PR closed since it went as an official PR on Zephyrproject

**Pull request closed without merging**